.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: Test
+on: [pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup python (${{ matrix.python-version }})
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Setup Python cache
+      uses: actions/cache@v2
+      id: python-cache
+      with:
+        path: venv
+        key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/requirements*.txt') }}
+        restore-keys: venv-${{ runner.os }}-${{ matrix.python-version }}-
+
+    - name: Setup Node cache
+      uses: actions/cache@v2
+      id: node-cache
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+    
+    - name: Install Node version
+      shell: bash -l {0}
+      run: nvm install
+
+    - name: Install Node dependencies
+      run:  make npm-install
+      if: steps.node-cache.outputs.cache-hit != 'true'
+
+    - name: Install python dependencies
+      run: make requirements-dev
+      if: steps.python-cache.outputs.cache-hit != 'true'
```

Comment:
This is cool, however I think `make test` will run those steps anyway?

Reply:
Yes, it does seem to but because the steps are not decoupled, there's no way to avoid that. At least this removes one repetition.

Reply:
👍